include/seqan3/utility/concept/one_of.hpp

```diff
@@ -0,0 +1,32 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::one_of.
+ * \author Hannes Hauswedell <hannes.hauswedell AT decode.is>
+ */
+
+#pragma once
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/concepts>
+
+namespace seqan3
+{
+/*!\interface   seqan3::one_of <>
+ * \ingroup utility_concept
+ * \tparam t    The query type to compare.
+ * \tparam ts   The reference types.
+ * \brief       Checks whether the list of reference types contains the query type.
+ */
+//!\cond
+template <typename t, typename ... ts>
+concept one_of = (std::same_as<t, ts> || ...);
```

Comment:
This rather looks like it should be part of the `type_traits/list` and `type_traits/pack` utility.

Reply:
Hm, but it is not for working with lists or packs. 

It does this:

```cpp
void foobar(one_of<std::string_view, std::string> auto & s)
{
  ....
}
```